example-github/src/main/java/feign/example/github/GitHubExample.java

```diff
@@ -46,69 +66,30 @@ public static void main(String... args) throws InterruptedException {
       System.out.println(contributor.login + " (" + contributor.contributions + ")");
     }
 
+    System.out.println("Now, let's cause an error.");
     try {
-      contributors = github.contributors("netflix", "some-unknown-project");
+      github.contributors("netflix", "some-unknown-project");
     } catch (GitHubClientError e) {
-      System.out.println(e.error.message);
+      System.out.println(e.getMessage());
     }
   }
 
-  interface GitHub {
-
-    @RequestLine("GET /repos/{owner}/{repo}/contributors")
-    List<Contributor> contributors(@Param("owner") String owner, @Param("repo") String repo);
-  }
-
-  static class Contributor {
-
-    String login;
-    int contributions;
-  }
-
-  static class ClientError {
-
-    String message;
-    List<Error> errors;
-  }
-
-  static class Error {
-    String resource;
-    String field;
-    String code;
-  }
-
   static class GitHubErrorDecoder implements ErrorDecoder {
 
     final Decoder decoder;
     final ErrorDecoder defaultDecoder = new ErrorDecoder.Default();
 
-    public GitHubErrorDecoder(Decoder decoder) {
+    GitHubErrorDecoder(Decoder decoder) {
       this.decoder = decoder;
     }
 
+    @Override
     public Exception decode(String methodKey, Response response) {
-      if (response.status() >= 400 && response.status() < 500) {
```

Comment:
I chose to take out this detail, as the example oversimplifies anyway. key story to tell is that we can use a decoder in the error decoder.